src/providers/xyz.js

```diff
@@ -43,8 +45,22 @@ XYZProvider.prototype = {
     d3.json(url, callback)
   },
 
-  invalidateCache: function () {
-    this.tileCache = {}
+  setReady: function () {
```

Comment:
Does this need to be a public method? I'd make it private otherwise (setURL is the public interface IMO)